### ISSUE TITLE
faction_id of npc corp using /corporations/{corporation_id}/

### ISSUE THREAD

**Issue by Blacksmoke16** [ccpgames/esi-issues#765]
# ATTENTION: DON'T SUBMIT SSO/CREST/XML RELATED ISSUES

# Bug

### Request
/corporations/{corporation_id}/ using NPC corp, example `1000063`

### Response
```
{
    "name": "Amarr Constructions",
    "ticker": "AC",
    "member_count": 85,
    "ceo_id": 3003877,
    "tax_rate": 0,
    "creator_id": 1,
    "description": "A low-profile, conservative construction company that likes to keep out of the lime-light. Amarr Constructions had its heyday when the Amarrians were first entering space and participated in the construction of the first star gates. The innovative side of the company is long gone, though it still retains much of its old power and majesty.",
    "home_station_id": 60006100,
    "shares": 15134828244
}
```

### Expected
Response should include the faction_id of the NPC corp, i.e. `500003`

**Comment by antihax** [comment]
faction_id is for faction warfare corporations.
see: `https://esi.tech.ccp.is/latest/corporations/1000180/`

```
{
  "name": "State Protectorate",
  "ticker": "SPROT",
  "member_count": 25113,
  "ceo_id": 3018995,
  "tax_rate": 0,
  "creator_id": 1,
  "description": "We are the State, and we have claimed our long-lost homeland. Now we are at war; united, whole and full of fire and purpose. The State calls you, capsuleer, for it needs your strength and your leadership to fend off the encroaching Gallente menace. The State will not fall. Join us. Fight. Conquer.",
  "url": "",
  "faction_id": 500001,
  "home_station_id": 60015069,
  "shares": 0
}
```

**Comment by Blacksmoke16** [comment]
Then in that case should it be `alliance_id`?  I think it can be argued either way.

**Comment by guiguilechat** [comment]
yes it should be alliance_id.

**Comment by tsuthers-ccpgames** [comment]
It should be faction ID. NPC corporations are not in alliances, which are a player construct, and are grouped to their NPC empires via their faction ID. The call in the OP should be returning a faction ID though, 500003 to be specific.

**Comment by CarbonAlabel** [comment]
@tsuthers-ccpgames this raises another problem: how are we supposed to know the difference between a player corp enrolled in FW, a regular NPC corp, and the FW NPC corp for a given faction?
It would be possible to figure it out with help of the /universe/factions/ endpoint, but it seems like far too much work for something that should be fairly simple.

I'd propose the use of separate properties to indicate the faction an NPC corp belongs to, and another one (`fw_faction_id`, perhaps) to indicate the faction for which player organizations are fighting for in factional warfare.

**Comment by lukasni** [comment]
The differences _are_ obvious from the ID range, but I'm not sure how official those are and whether they're guaranteed to stay reliable.

**Comment by CarbonAlabel** [comment]
True, but you would still need to reference the /universe/factions/ endpoint to see if the NPC corporation is **the** FW corporation. This matters, as you can only fight members of that single NPC corporation under FW rules.

Aside from that, I believe one of the guiding principles of ESI is still to make its usage depend as little as possible on the use of external information sources, and knowing which ID range NPC corporations are located in would be one of those.

**Comment by Seavert** [comment]
Current `faction_id` can be renamed to `warfare_faction_id`.
And new `faction_id` can be added as requested attribute of NPC corps.

**Comment by guiguilechat** [comment]
so … Is there a way to find the faction an agent belongs to ?

**Comment by FFX3** [comment]
Has there been any improvements?

I'm currently using "https://[esi.evetech.net/latest/universe/factions/](https://esi.evetech.net/latest/universe/factions/)".  But it only has 27 corps.

Hyasyoda is one example of a corp I can't look up.

**Comment by fc-pinky** [comment]
With the compatibility-date of 2026-07-21, the `/corporation/{corporation_id}` not gives the field as `enlisted_faction_id`, to make more clear that the field is about Factional Warfare.

The faction an NPC belongs to can be found in the SDE, and won't be available via ESI. Hopefully that addresses all the points in this ticket. If not, please create a new ticket with a clear use-case what the issue is.
